pallets/schemas/src/lib.rs

```diff
@@ -184,16 +188,24 @@ pub mod pallet {
 	pub(super) type CurrentSchemaIdentifierMaximum<T: Config> =
 		StorageValue<_, SchemaId, ValueQuery>;
 
+	/// Storage for message schema info struct data
+	/// - Key: Schema Id
+	/// - Value: [`SchemaInfo`](SchemaInfo)
+	#[pallet::storage]
+	#[pallet::getter(fn get_schema_info)]
+	pub(super) type SchemaInfos<T: Config> =
```

Comment:
So I've looked into [CountedStorageMap](https://github.com/paritytech/polkadot-sdk/blob/18257373b313fcfad4f7a2ab2cb02cb8bfc5b8f1/substrate/frame/support/src/storage/types/counted_map.rs#L198-L203) again. The implementation is very generic and not efficient. For every insert it checks if the key exists and then does the increment. Since for our use-case we know that every key would be new so keeping them separate is more efficient.

Reply:
Sounds good. Let's keep it as is for now.